.github/workflows/deployments.yml

```diff
@@ -404,13 +411,202 @@ jobs:
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - name: Set up context for buildx
+        run: |
+          docker context create builder_context
+
+      - name: Set up buildx runner
+        uses: docker/setup-buildx-action@v3
+        with:
+          endpoint: builder_context
+          version: v0.19.0
+          driver-opts: |
+            image=moby/buildkit:v0.19.0
+
+      - name: Create multi-arch images
+        id: stitch
+        run: |
+          set -euo pipefail
+
+          DIGESTS_JSON='${{ needs.digest_config.outputs.json }}'
+          EXT_JSON='${{ needs.ext_config.outputs.json }}'
+          echo "$DIGESTS_JSON" | jq .
+          echo "$EXT_JSON" | jq .
+
+
+          stitched_output="{}"
+
+          echo "Creating stitched manifest for cuda-quantum-devdeps..."
+          for cuda in 11.8 12.0; do
+            cuda_major="${cuda%%.*}"
+            key="arm64-cu${cuda}-ext"
+            arm64_image=$(echo "$EXT_JSON" | jq -r ".image_hash[\"$key\"]")
+            echo "CUDA $cuda arm64 image: $arm64_image"
+
+            key="amd64-cu${cuda}-ext"
+            amd64_image=$(echo "$EXT_JSON" | jq -r ".image_hash[\"$key\"]")
+            echo "CUDA $cuda amd64 image: $amd64_image"
+
+            stitched_tag="ghcr.io/nvidia/cuda-quantum-dev:cu${cuda}-ext"
+
+            refs="$arm64_image $amd64_image"
+
+            docker buildx imagetools create --tag "$stitched_tag" $refs
+            digest=$(docker buildx imagetools inspect "$stitched_tag" | awk '/^Digest:/ { print $2 }')
+            stitched_key="devdeps-cu${cuda_major}"
+            stitched_output=$(echo "$stitched_output" | jq --arg key "$stitched_key" --arg val "$digest" '. + {($key): $val}')
+          done
+
+          echo "Creating stitched manifest for cuda-quantum and cuda-quantum-dev..."
+          for image in image dev-image; do
+            for cuda in 11.8 12.0; do
+              echo "Processing CUDA $cuda..."
+              refs=""
+              for arch in amd64 arm64; do
+                key="${arch}-cu${cuda}-${image}"
+                echo "Looking up key: $key"
+
+                digest=$(echo "$DIGESTS_JSON" | jq -r ".digest[\"$key\"]")
+                if [[ -z "$digest" || "$digest" == "null" ]]; then
+                  echo "::error::Missing digest for key $key"
+                  exit 1
+                fi
+
+                # Registry must match where the digest was pushed
+                if [[ "$image" == "dev-image" ]]; then
+                  registry="ghcr.io/nvidia/cuda-quantum-dev"
+                else
+                  registry="ghcr.io/nvidia/cuda-quantum"
+                fi
+
+                refs="$refs $registry@$digest"
+              done
+
+              cuda_major="${cuda%%.*}"
+
+              if [[ "$image" == "dev-image" ]]; then
+                stitched_tag="ghcr.io/nvidia/cuda-quantum-dev:cu${cuda_major}-latest-base"
```

Comment:
Similar comment here for the tags; latest is only used for the main branch, not for all other branches that have publishing support (release branches, experimental branches, ...).

Reply:
why not use:
```
stitched_tag="ghcr.io/nvidia/cuda-quantum-dev:cu${cuda}-ext-${{ github.ref_name }}"
```

This point in the deployment shouldn't be used by a PR, but instead ran by you so `ref_name` shouldn't have any of the branch information.